ReactApp/src/misc/api-fetch-setup.js

```diff
@@ -141,3 +142,27 @@ function filterEndpointsMiddleware(options, next) {
 	}
 	return next(options);
 }
+
+/**
+ * Middleware to modify media upload requests.
+ *
+ * This middleware intercepts requests to the media endpoint and conditionally
+ * removes the 'post' field if its value is '-1', which is used for draft posts.
+ */
+function mediaUploadMiddleware(options, next) {
+	if (options.path.startsWith('/wp/v2/media') && options.method === 'POST') {
+		if (options.body instanceof FormData) {
+			const newFormData = new FormData();
+
+			options.body.forEach((value, key) => {
+				if (key === 'post' && value === '-1') {
+					return;
+				}
+				newFormData.append(key, value);
+			});
+
+			options.body = newFormData;
+		}
+	}
```

Comment:
I didn't test this suggested code, but would it make sense to short this to avoid nesting and iterating? 

```suggestion
	if (
		options.path.startsWith('/wp/v2/media') &&
		options.method === 'POST' &&
		options.body instanceof FormData &&
		options.body.get('post') === '-1'
	) {
		options.body.delete('post');
	}

```